### PR TITLE
Extend support for streaming datasets that use glob.glob

### PR DESCRIPTION
**Pull request by albertvillanova**
This PR extends the support in streaming mode for datasets that use `glob`, by patching the function `glob.glob`.

Related to #2880, #2876, #2874

### CODE REVIEW

File: tests/test_streaming_download_manager.py [hunk 1]
```diff
@@ -219,6 +220,46 @@ def test_xopen_remote():
...
+    if input_path.startswith("tmp_path"):
+        input_path = input_path.replace("/", os.sep).replace("tmp_path", str(tmp_path))
+        expected_paths = [str(tmp_path / file) for file in expected_paths]
+        for file in ["file1.txt", "file2.txt", "README.md"]:
+            (tmp_path / file).touch()
+        output_path = sorted(xglob(input_path))
+    else:
+        dummy_registry = datasets.utils.streaming_download_manager.fsspec.registry.target.copy()
+        dummy_registry["mock"] = DummyTestFS
+        with patch.dict(datasets.utils.streaming_download_manager.fsspec.registry.target, dummy_registry):
```
Comment (lhoestq) [inline tests/test_streaming_download_manager.py:258]: This could be a pytest fixture ?

File: tests/test_streaming_download_manager.py [hunk 2]
```diff
@@ -219,6 +220,46 @@ def test_xopen_remote():
...
+            output_path = sorted(xglob(input_path))
+    assert output_path == expected_paths
```
Comment (lhoestq) [inline tests/test_streaming_download_manager.py:260]: `output_path` -> `output_paths`

**Review by lhoestq — APPROVED** [review]
Awesome thanks !

File: tests/test_streaming_download_manager.py (hunk 1 above)
Comment (albertvillanova) [reply tests/test_streaming_download_manager.py]: Done!

**Review by lhoestq — APPROVED** [review]
thank you :)

**Pull request merged**